Sources/BrowserServicesKit/PrivacyConfig/AppPrivacyConfiguration.swift

```diff
@@ -88,14 +96,71 @@ public struct AppPrivacyConfiguration: PrivacyConfiguration {
         default: return false
         }
     }
+    
+    private func rolloutEnabled(subfeature: any PrivacySubfeature,
+                                rollouts: [PrivacyConfigurationData.PrivacyFeature.Feature.Rollout],
+                                randomizer: (Range<Double>) -> Double) -> Bool {
+        let defsPrefix = "config.\(subfeature.parent.rawValue).\(subfeature.rawValue)"
+        if userDefaults.bool(forKey: "\(defsPrefix).\(Constants.enabledKey)") {
+            return true
+        }
+        
+        var willEnable = false
+        if let rolloutSize = userDefaults.value(forKey: "\(defsPrefix).\(Constants.lastRolloutCountKey)") as? Int {
+            guard rolloutSize != rollouts.count else { return false }
+            // Sanity check as we need at least two values to compute the new probability
+            guard rollouts.count > 1 else { return false }
+            
+            // If the user has seen the rollout before, and the rollout count has changed
+            // Try again with the new probability
+            let y = rollouts[rollouts.count - 1].percent
+            let x = rollouts[rollouts.count - 2].percent
+            let prob = (y - x) / (100.0 - y)
```

Comment:
If we somehow have a case where y is `100` this would lead to a divide by 0. That's real edge case territory though 🙂

Reply:
I misread the spec here. We're supposed to subtract x from 100 here.